packages/block-library/src/image/index.php

```diff
@@ -0,0 +1,42 @@
+<?php
+/**
+ * Server-side rendering of the `core/image` block.
+ *
+ * @package gutenberg
+ */
+
+/**
+ * Renders the `core/image` block on the server,
+ * adding a data-id attribute to the element if core/gallery has added on pre-render.
+ *
+ * @param  array $attributes The block attributes.
+ * @param  array $content    The block content.
+ * @return string            Returns the block content with the data-id attribute added.
+ */
+function render_block_core_image( $attributes, $content ) {
+	if ( isset( $attributes['data-id'] ) ) {
+		// Add the data-id="$id" attribute to the img element
+		// to provide backwards compatibility for the Gallery Block,
+		// which now wraps Image Blocks within innerBlocks.
+		// The data-id attribute is added in a core/gallery `render_block_data` hook.
+		$data_id_attribute = 'data-id="' . esc_attr( $attributes['data-id'] ) . '"';
+		if ( ! strpos( $content, $data_id_attribute ) ) {
```

Comment:
There should be an explicit check for `false`, otherwise a match at position 0 of `$content` will satisfy the if-statement's condition:

```php
if ( false === strpos( $content, $data_id_attribute ) ) {
```

(Of course, semantically in this particular function we should never have a match at 0, but this deserves a patch. :) cc @ramonjd)

Reply:
Thanks @mcsf 

I'll get a PR up today.

Reply:
Thanks, and thanks especially for humouring my random comment three months after the fact! 😅

Reply:
I'll be here in another 3 months :trollface: Chat then. 

Haha. Seriously, always happy to take on advice for improvements.